### PR TITLE
Matched RDP (dp), audio interface (ai), EEPROM (conteep) files, and some misc files

### PR DESCRIPTION
**Pull request by Mr-Wiseguy**

### CODE REVIEW

File: src/io/conteeplongread.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include "PR/os_internal.h"
+#include "controller.h"
+
+s32 osEepromLongRead(OSMesgQueue *mq, u8 address, u8 *buffer, int length) {
+    s32 ret;
+    ret = 0;
```
Comment (Thar0) [inline src/io/conteeplongread.c:6]:
```suggestion
    s32 ret = 0;
```

File: src/io/conteeplongwrite.c [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#include "PR/os_internal.h"
+#include "controller.h"
+
+s32 osEepromLongWrite(OSMesgQueue *mq, u8 address, u8 *buffer, int length) {
+    s32 ret;
+    ret = 0;
```
Comment (Thar0) [inline src/io/conteeplongwrite.c:6]:
```suggestion
    s32 ret = 0;
```

File: src/io/conteepprobe.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#include "controller.h"
+#include "siint.h"
+
+s32 osEepromProbe(OSMesgQueue *mq) {
+    s32 ret;
+    u16 type;
+    OSContStatus sdata;
+
+    ret = 0;
```
Comment (Thar0) [inline src/io/conteepprobe.c:9]:
```suggestion
    s32 ret = 0;
    u16 type;
    OSContStatus sdata;
```

File: src/io/conteepread.c [hunk 4]
```diff
@@ -0,0 +1,104 @@
+#include "macros.h"
+#include "PR/rcp.h"
+#include "controller.h"
+#include "siint.h"
+
+static void __osPackEepReadData(u8 address);
+OSPifRam __osEepPifRam ALIGNED(16);
+s32 __osEepromRead16K;
+s32 osEepromRead(OSMesgQueue *mq, u8 address, u8 *buffer)
```
Comment (Thar0) [inline src/io/conteepread.c:9]:
```suggestion
s32 __osEepromRead16K;

s32 osEepromRead(OSMesgQueue *mq, u8 address, u8 *buffer)
```

File: src/io/conteepread.c [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+{
+	s32 ret;
+	int i;
+	u16 type;
+	u8 *ptr;
+	OSContStatus sdata;
+	__OSContEepromFormat eepromformat;
+
+	ret = 0;
+	i = 0;
```
Comment (Thar0) [inline src/io/conteepread.c:19]:
```suggestion
	s32 ret = 0;
	s32 i = 0;
	u16 type;
	u8 *ptr;
	OSContStatus sdata;
	__OSContEepromFormat eepromformat;

```
Disregard the type change if it's from mdebug

File: src/io/conteepread.c [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	__osSiRelAccess();
+	return ret;
+}
+
+static void __osPackEepReadData(u8 address) {
+	u8 *ptr;
+	__OSContEepromFormat eepromformat;
+	int i;
+
+	ptr = (u8 *)&__osEepPifRam.ramarray;
```
Comment (Thar0) [inline src/io/conteepread.c:87]:
```suggestion
	u8 *ptr = (u8 *)&__osEepPifRam.ramarray;
	__OSContEepromFormat eepromformat;
	s32 i;

```
Same for this type change

File: src/io/conteepwrite.c [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    s32 ret;
+    int i;
+    u16 type;
+    u8 *ptr;
+    __OSContEepromFormat eepromformat;
+    OSContStatus sdata;
+    u8 temp[8];
+
+    ret = 0;
+    ptr = (u8 *)&__osEepPifRam.ramarray;
```
Comment (Thar0) [inline src/io/conteepwrite.c:18]:
```suggestion
    s32 ret = 0;
    s32 i;
    u16 type;
    u8 *ptr = (u8 *)&__osEepPifRam.ramarray;
    __OSContEepromFormat eepromformat;
    OSContStatus sdata;
    u8 temp[8];

```

File: src/io/conteepwrite.c [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+    __osSiRelAccess();
+    return ret;
+}
+
+static void __osPackEepWriteData(u8 address, u8 *buffer) {
+    u8 *ptr;
+    __OSContEepromFormat eepromformat;
+    int i;
+    ptr = (u8 *)&__osEepPifRam.ramarray;
```
Comment (Thar0) [inline src/io/conteepwrite.c:84]:
```suggestion
    u8 *ptr = (u8 *)&__osEepPifRam.ramarray;
    __OSContEepromFormat eepromformat;
    s32 i;
```

File: src/io/conteepwrite.c [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+
+s32 __osEepStatus(OSMesgQueue *mq, OSContStatus *data) {
+    s32 ret;
+    int i;
+    u8 *ptr;
+    __OSContRequesFormat requestformat;
+
+    ret = 0;
+    ptr = (u8 *)__osEepPifRam.ramarray;
```
Comment (Thar0) [inline src/io/conteepwrite.c:115]:
```suggestion
    s32 ret = 0;
    s32 i;
    u8 *ptr = (u8 *)__osEepPifRam.ramarray;
    __OSContRequesFormat requestformat;
```

File: src/io/conteepwrite.c [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+    
+    for (i = 0; i < ARRLEN(__osEepPifRam.ramarray) + 1; i++) {
+        __osEepPifRam.ramarray[i] = 0;
+    }
+
+    __osEepPifRam.pifstatus = CONT_CMD_EXE;
+    ptr = (u8 *)__osEepPifRam.ramarray;
+    for (i = 0; i < 4; i++) //zero 4 bytes?
+        *ptr++ = 0;
```
Comment (Thar0) [inline src/io/conteepwrite.c:124]:
```suggestion
    for (i = 0; i < 4; i++) { //zero 4 bytes?
        *ptr++ = 0;
    }
```

File: src/io/devmgr.c [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+};
+
+// Is this cheating? Too bad!
+#define LOAD_PHYSICAL_ADDR(out, addr)                    \
+    __asm__ __volatile__("lui %0, %1"                    \
+                         : "=r"(out)                     \
+                         : "i"(PHYS_TO_K1(addr) >> 16)); \
+    __asm__ __volatile__("ori %0, %1, %2"                \
+                         : "=r"(out)                     \
+                         : "r"(out), "i"(PHYS_TO_K1(addr) & 0xFFFF));
```
Comment (Thar0) [inline src/io/devmgr.c:17]:
```suggestion
struct test
{
    u32 test;
};

// Is this cheating? Too bad!
#define LOAD_PHYSICAL_ADDR(out, addr)                    \
    __asm__ __volatile__("lui %0, %1"                    \
                         : "=r"(out)                     \
                         : "i"(PHYS_TO_K1(addr) >> 16)); \
    __asm__ __volatile__("ori %0, %1, %2"                \
                         : "=r"(out)                     \
                         : "r"(out), "i"(PHYS_TO_K1(addr) & 0xFFFF));
```
Is any of this still needed?

File: src/io/dp.c [hunk 12]
```diff
@@ -0,0 +1,13 @@
+#include "PR/os_internal.h"
+#include "PR/rcp.h"
+
+// TODO: this comes from a header
+#ident "$Revision: 1.17 $"
+
+int __osDpDeviceBusy(void) {
+    register u32 stat;
+    stat = IO_READ(DPC_STATUS_REG);
```
Comment (Thar0) [inline src/io/dp.c:9]:
```suggestion
    register u32 stat = IO_READ(DPC_STATUS_REG);
```

File: src/io/dp.c [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+    if (stat & DPC_STATUS_DMA_BUSY)
+        return 1;
```
Comment (Thar0) [inline src/io/dp.c:11]:
```suggestion
    if (stat & DPC_STATUS_DMA_BUSY) {
        return 1;
    }
```

File: src/os/gethwinterrupt.c [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+    s32 (*handler)(void);
+    void *stackEnd;
+};
+
+extern struct __osHwInt __osHwIntTable[];
+
+void __osGetHWIntrRoutine(OSHWIntr interrupt, s32 (**handler)(void), void **stackEnd) {
+    *handler = __osHwIntTable[interrupt].handler;
+    *stackEnd = __osHwIntTable[interrupt].stackEnd;
+}
```
Comment (Thar0) [inline src/os/gethwinterrupt.c:13]:
```suggestion
}

```

**Review by Thar0 — CHANGES_REQUESTED** [review]
Mostly little formatting things

File: src/io/aisetnextbuf.c [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+    bptr = bufPtr;
+
+	if (hdwrBugFlag != 0) {
+		bptr -= 0x2000;
+	}
+
+	if ((((u32)bufPtr + size) & 0x1fff) == 0) {
+		hdwrBugFlag = 1;
+	} else {
+		hdwrBugFlag = 0;
```
[the remark this answers is not in the document]
Comment (Mr-Wiseguy) [reply src/io/aisetnextbuf.c:25]: Agreed

File: src/io/conteepread.c (hunk 5 above)
Comment (Mr-Wiseguy) [reply src/io/conteepread.c:19]: It's int in the mdebug.

File: src/io/devmgr.c (hunk 11 above)
Comment (Mr-Wiseguy) [reply src/io/devmgr.c:17]: Oops, left that in from how I had it previously matched. Not needed.

**Comment by Mr-Wiseguy** [comment]
Addressed all of the suggestions.

**Review by Thar0 — APPROVED** [review]

**Pull request merged**